### PR TITLE
Storage

### PR DESCRIPTION
**Pull request by Paskab012**
### Adding local storage to the project 
- This action will enable the user credentials to be automatically fill in 
- Those credentials are kept into that local storage

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -13,3 +14,135 @@ menuLinks[1].addEventListener('click', mobileMenuT);
...
+            `[data-popup-start='${popupId}',]`,
+          );
+          popupStart.scrollIntoView();
+        });
+      });
+    }
+  });
+});
+
+function inputSave() {
```
Comment (andersonlebon) [inline app.js:139]: - [ ] Kindly read and understand perfectly the project requirements. I know you can do it :muscle: :muscle:. Watch this [video](https://www.youtube.com/watch?v=C3tiI3akFq8), or [this](https://www.youtube.com/watch?v=9n3qAeaBMGU). They can help you.
requirement

File: app.js [hunk 2]
```diff
@@ -13,3 +14,135 @@ menuLinks[1].addEventListener('click', mobileMenuT);
...
+  const name = document.getElementById('username').value;
+  const email = document.getElementById('email').value;
+  const message = document.getElementById('msg').value;
+
+  localStorage.setItem('username', name);
+  localStorage.setItem('email', email);
+  localStorage.setItem('msg', message);
+}
+inputSave();
```
Comment (andersonlebon) [inline app.js]: - [x] Kindly read and understand perfectly the project requirements. I know you can do it :muscle: :muscle:. Watch this [video](https://www.youtube.com/watch?v=C3tiI3akFq8), or [this](https://www.youtube.com/watch?v=9n3qAeaBMGU). They can help you.
requirement

File: app.js [hunk 3]
```diff
@@ -13,3 +14,135 @@ menuLinks[1].addEventListener('click', mobileMenuT);
...
+      });
+    }
+  });
+});
+
+function inputSave() {
+  const name = document.getElementById('username').value;
+  const email = document.getElementById('email').value;
+  const message = document.getElementById('msg').value;
+
```
Comment (andersonlebon) [inline app.js]: - [x] Kindly store the form data for all inputs in a single object, saved to a single local storage registry.

Comment (andersonlebon) [inline app.js]: - [x] Kidly add the functionality when the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
Hi @KABIKA681 and @Elerqsousy 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- You  are following GitHub actions :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

-----

- [x] I think it would be better to replace the **" write your message here "** message in the `Textarea` tag with the `Placeholder` attribute to make your form looks professional :100: 
textarea

File: storage.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    nameField.value = formData.fullName;
+    emailField.value = formData.emailAddress;
+    messageField.value = formData.message;
+  }
+}
+
+window.addEventListener('load', inputForm);
+
+for (let n = 0; n < contactCard.length - 1; n += 1) {
+  contactCard[n].addEventListener('change', () => {
```
Comment (Gambit142) [inline storage.js]: - [ ] You are almost through. Presently, the information inputted in the `textarea` is not stored in the `localStorage`. Line 26 should look like this `for (let n = 0; n < contactCard.length; n += 1)` not the way it is.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @KABIKA681 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Kindly check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @KABIKA681, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
**Hi @team**, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**